src/EditorControls.js

```diff
@@ -228,19 +228,20 @@ class EditorControls extends Component {
         break;
 
       case EDITOR_ACTIONS.DELETE_TRANSFORM:
-        if (isNumeric(payload.transformIndex)) {
-          for (let i = 0; i < graphDiv.data.length; i++) {
-            if (graphDiv.data[i].uid === payload.traceUid) {
-              graphDiv.data[i].transforms.splice(payload.transformIndex, 1);
-              if (this.props.onUpdate) {
-                this.props.onUpdate(
-                  graphDiv.data.slice(),
-                  graphDiv.layout,
-                  graphDiv._transitionData._frames
-                );
-              }
-              break;
-            }
+        if (
+          isNumeric(payload.transformIndex) &&
+          payload.traceIndex < graphDiv.data.length
```

Comment:
i'm not sure I understand this line: `payload.traceIndex < graphDiv.data.length`

before we used to loop and find the trace in data that matched our transform index. how is this replaced now with the line above?

do we not need to loop, like we used to but have the comparison be:
if `(payload.transformIndex === graphDiv._fullData[i].index)` in place of `if (graphDiv.data[i].uid === payload.traceUid)`

Reply:
I figure we looped in order to find the UID, no we go by index alone. Why did we loop in the first place, when we could just do graphDiv.data.find(...)?..

Reply:
payload.transformIndex, indicates a _fullData index?

indexes are not 1:1, an index in data could have a different index in _fullData

Reply:
plz check the discussion in plotly_js slack:

nicolas [13:40]
we're trying to match up from `data` not `fullData`


etienne [13:41]
`fullData[i]._fullInput.index` will be the index in `gd.data`
nicolas [13:42]
ok so we might be able to go from `if (trace.uid === fullData[i]._fullInput._input.uid) {`
to `if (theIndex === fullData[i]._fullInput._input.index) {`
assuming `theIndex` is the position of `trace` in the `data` array?
etienne [13:44]
yeah, that should work

Reply:
ok sure, yes them it looks like you don't need that conditional, 
thanks!